values/time.go

```diff
@@ -211,29 +227,81 @@ func (d Duration) String() string {
 		weeks := d.nsecs
 
 		if weeks > 0 {
-			writeInt(&sb, weeks, "w")
+			values = append(values, ast.Duration{
+				Magnitude: weeks * scale,
+				Unit:      ast.WeekUnit,
+			})
 		}
 		if days > 0 {
-			writeInt(&sb, days, "d")
+			values = append(values, ast.Duration{
+				Magnitude: days * scale,
+				Unit:      ast.DayUnit,
+			})
 		}
 		if hours > 0 {
-			writeInt(&sb, hours, "h")
+			values = append(values, ast.Duration{
+				Magnitude: hours * scale,
+				Unit:      ast.HourUnit,
+			})
 		}
 		if mins > 0 {
-			writeInt(&sb, mins, "m")
+			values = append(values, ast.Duration{
+				Magnitude: mins * scale,
+				Unit:      ast.MinuteUnit,
+			})
 		}
 		if secs > 0 {
-			writeInt(&sb, secs, "s")
+			values = append(values, ast.Duration{
+				Magnitude: secs * scale,
+				Unit:      ast.SecondUnit,
+			})
 		}
 		if msecs > 0 {
-			writeInt(&sb, msecs, "ms")
+			values = append(values, ast.Duration{
+				Magnitude: msecs * scale,
+				Unit:      ast.MillisecondUnit,
+			})
 		}
 		if usecs > 0 {
-			writeInt(&sb, usecs, "us")
+			values = append(values, ast.Duration{
+				Magnitude: usecs * scale,
+				Unit:      ast.MicrosecondUnit,
+			})
 		}
 		if nsecs > 0 {
-			writeInt(&sb, nsecs, "ns")
+			values = append(values, ast.Duration{
+				Magnitude: nsecs * scale,
+				Unit:      ast.NanosecondUnit,
+			})
+		}
+	}
+	return values
+}
+
+func (d Duration) String() string {
+	values := d.AsValues()
+	if len(values) == 0 {
+		return "0ns"
+	}
+
+	var buf []byte
+	writeInt := func(sb *strings.Builder, v int64, unit string) {
```

Comment:
I thought I reviewed very similar code somewhere else.  Where was it?  If there is some duplication here, maybe it could be removed somehow?  If not, no big deal.

Reply:
I think it was taken from the month addition PR since I needed it here.